tests/test_app_client.py

```diff
@@ -175,13 +324,33 @@ async def test_get_robot_part_logs(self, service: MockApp):
         async with ChannelFor([service]) as channel:
             client = AppClient(channel, METADATA, ID)
             log_entries = await client.get_robot_part_logs(
-                robot_part_id=ID, filter=FILTER, errors_only=ERRORS_ONLY, num_log_entries=NUM_LOGS
+                robot_part_id=ID, filter=FILTER, errors_only=ERRORS_ONLY, num_log_entries=NUM
             )
             assert service.robot_part_id == ID
             assert service.filter == FILTER
             assert service.errors_only == ERRORS_ONLY
             assert [log_entry.proto for log_entry in log_entries] == LOG_ENTRIES
 
+    @pytest.mark.asyncio
+    async def test_tail_robot_part_logs(self, service: MockApp):
+        async with ChannelFor([service]) as channel:
+            client = AppClient(channel, METADATA, ID)
+            logs_stream = await client.tail_robot_part_logs(robot_part_id=ID, errors_only=ERRORS_ONLY, filter=FILTER)
+            [logs async for logs in logs_stream]
```

Comment:
logs_stream, the returned variable from tail_robot_part_logs is an async iterator. Because of the way the method is written/defined, the mock server does not execute properly unless we actually iterate through the logs_stream (forcing its __anext__() method to be called). This means that without line 338 (the list comprehension thing), the following assertion statements all fail because the mock server never gets to the lines that actually set its members to the parameters we pass in the method call.

Reply:
Can we add a short comment explaining this? I don't think it's very intuitive.